### PR TITLE
KEYCLOAK-6519 Documentation for theme resource provider

### PR DESCRIPTION
**Pull request by stianst**

### CODE REVIEW

File: server_development/topics/themes-resources.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[[_theme_resource]]
+= Theme Resources
+
+Templates and resources are by default loaded from the themes, but it is possible to provide additional templates
+and resources through the ThemeResourceSPI.
+
```
Comment (matthewhelmke) [inline server_development/topics/themes-resources.adoc]: Templates and resources are loaded from the themes by default,

**Review by matthewhelmke — DISMISSED** [review]
Small change suggested. Not a blocker for merging.

**Comment by stianst** [comment]
@matthewhelmke Updated again. Marek made me improve this stuff ;)

**Comment by matthewhelmke** [comment]
Couple errors, @stianst

**Comment by stianst** [comment]
Yah, the auth spi section is not included in product yet as it's unsupported. Changed it so the example use-case of auth-spi is only shown in community docs

**Review by matthewhelmke — DISMISSED** [review]
Very good. Thanks, @stianst and also thanks to @mposolda for some excellent suggestions for improvement.

File: server_development/topics/auth-spi.adoc [hunk 2]
```diff
@@ -374,8 +374,8 @@ getReferenceCategory() is just a category the Authenticator belongs to.
 ==== Adding Authenticator Form
 
 Keycloak comes with a Freemarker <<_themes,theme and template engine>>.
-The createForm() method you called within authenticate() of your Authenticator class, builds an HTML page from a file within your login theme: secret-question.ftl.
-This file should be placed in the login theme with all the other .ftl files you see for login. 
+The createForm() method you called within authenticate() of your Authenticator class, builds an HTML page from a file within your login theme: `secret-question.ftl`.
+This file should be made available by creating a custom <<_theme_resource,Theme Resource Provider>>.
```
Comment (mposolda) [inline server_development/topics/auth-spi.adoc]: @stianst The last sentence could be possibly changed, that now you don't necessarily Theme Resource Provider anymore? WDYT?

**Review by mposolda — COMMENTED** [review]
Left one comment in auth-spi.adoc file. The rest looks great to me.

**Comment by stianst** [comment]
@mposolda changed the line you commented on to "This file should be added to the `theme-resources-templates` in your JAR, see <<_theme_resource,Theme Resource Provider>> for more details."

**Review by mposolda — APPROVED** [review]
Nice, thanks!

**Pull request merged**